src/operators/source/ogr_source.cpp

```diff
@@ -92,3 +90,90 @@ void OGRSourceOperator::writeSemanticParameters(std::ostringstream &stream) {
+
+    if(OGRSourceDatasets::hasJsonParameter(layer_json, dataset_json, "default"))
+        constructed_params["default"] = OGRSourceDatasets::getJsonParameter(layer_json, dataset_json, "default");
+
+    if(OGRSourceDatasets::hasJsonParameter(layer_json, dataset_json, "force_ogr_time_filter"))
+        constructed_params["force_ogr_time_filter"] = OGRSourceDatasets::getJsonParameter(layer_json, dataset_json, "force_ogr_time_filter");
+
+    if(OGRSourceDatasets::hasJsonParameter(layer_json, dataset_json, "on_error"))
+        constructed_params["on_error"] = OGRSourceDatasets::getJsonParameter(layer_json, dataset_json, "on_error");
+
+    // columns
+    Json::Value columns(Json::ValueType::objectValue);
+    Json::Value layer_columns   = layer_json["columns"];
+    Json::Value dataset_columns = dataset_json["columns"];
+
+    if(OGRSourceDatasets::hasJsonParameter(layer_columns, dataset_columns, "x"))
+        columns["x"] = OGRSourceDatasets::getJsonParameter(layer_columns, dataset_columns, "x");
+    if(OGRSourceDatasets::hasJsonParameter(layer_columns, dataset_columns, "y"))
+        columns["y"] = OGRSourceDatasets::getJsonParameter(layer_columns, dataset_columns, "y");
+
+    if(OGRSourceDatasets::hasJsonParameter(layer_columns, dataset_columns, "time1"))
+        columns["time1"] = OGRSourceDatasets::getJsonParameter(layer_columns, dataset_columns, "time1");
+    if(OGRSourceDatasets::hasJsonParameter(layer_columns, dataset_columns, "time2"))
+        columns["time2"] = OGRSourceDatasets::getJsonParameter(layer_columns, dataset_columns, "time2");
+
+    columns["textual"]            = params["textual"];
+    columns["numeric"]            = params["numeric"];
+    constructed_params["columns"] = columns;
+
+    // provenance information
+    Json::Value dataset_provenance = dataset_json["provenance"];
+    Json::Value layer_provenance   = dataset_json["provenance"];
+    Json::Value provenanceInfo;
+    if(OGRSourceDatasets::hasJsonParameter(layer_provenance, dataset_provenance, "citation"))
+        provenanceInfo["citation"] = OGRSourceDatasets::getJsonParameter(layer_provenance, dataset_provenance, "citation");
+
+    if(OGRSourceDatasets::hasJsonParameter(layer_provenance, dataset_provenance, "license"))
+        provenanceInfo["license"] = OGRSourceDatasets::getJsonParameter(layer_provenance, dataset_provenance, "license");
+
+    if(OGRSourceDatasets::hasJsonParameter(layer_provenance, dataset_provenance, "uri"))
+        provenanceInfo["uri"] = OGRSourceDatasets::getJsonParameter(layer_provenance, dataset_provenance, "uri");
+
+    constructed_params["provenance"] = provenanceInfo;
+
+    return constructed_params;
+}
+
+Json::Value OGRSourceOperator::getTimeFormatJson(Json::Value &layer_json, Json::Value &dataset_json, const std::string &time_format_name){
+    Json::Value time_format_source;
+    if(layer_json.isMember(time_format_name)) {
```

Comment:
why note use `OGRSourceDatasets::getJsonParameter` here?

Reply:
Missed that one. Pushed an adjustment.

Reply:
Just realized too late that the whole function is obsolete. Thought you mean only the if-else. I will adjust it.